### PR TITLE
Restructure acceptance test folders

### PR DESCRIPTION
**Pull request by yeahfro**
This change restructure acceptance tests which will assist us with finding a repeatable pattern.

screen shot 2017-09-20 at 15 56 07

### CODE REVIEW

File: test/acceptance/nightwatch.conf.js [hunk 1]
```diff
@@ -1,18 +1,24 @@
 require('dotenv').config()
 const seleniumServer = require('selenium-server')
 const chromeDriver = require('chromedriver')
+const { readdirSync } = require('fs')
+const { join } = require('path')
+const { without } = require('lodash')
+
+const features = readdirSync(join(__dirname, 'features'))
```
Comment (feedmypixel) [inline test/acceptance/nightwatch.conf.js]: could you just have `readdirSync('./features')` ?

File: test/acceptance/nightwatch.conf.js [hunk 2]
```diff
@@ -1,18 +1,24 @@
...
+const apps = without(features, 'step-definitions')
```
Comment (feedmypixel) [inline test/acceptance/nightwatch.conf.js]: not sure if `apps` is the right name here? maybe `pageObjects`?

Comment (feedmypixel) [inline test/acceptance/nightwatch.conf.js]: could you be explicit in getting `pageObject`? by using a filter and only gathering `page-objects` paths?

File: test/acceptance/nightwatch.conf.js [hunk 3]
```diff
@@ -1,18 +1,24 @@
...
-    '--require', 'test/acceptance/features/step_definitions',
+    '--require', 'test/acceptance/features',
     '--format', 'json:cucumber/reports/tests.json',
     process.env.FEATURES_FOLDER || 'test/acceptance/features',
   ],
 })
 
 module.exports = {
-  page_objects_path: 'test/acceptance/page_objects',
+  page_objects_path: apps.map((app) => { return `test/acceptance/features/${app}/page-objects` }),
```
Comment (feedmypixel) [inline test/acceptance/nightwatch.conf.js]: if you create an array of `page-objects` paths above then you shouldn't have to do this

File: test/acceptance/nightwatch.conf.js [hunk 4]
```diff
@@ -1,18 +1,24 @@
 require('dotenv').config()
 const seleniumServer = require('selenium-server')
 const chromeDriver = require('chromedriver')
+const { readdirSync } = require('fs')
+
+const pageObjects =
+  readdirSync('./features')
```
Comment (feedmypixel) [inline test/acceptance/nightwatch.conf.js]: so this needs to be `test/acceptance/features` as it currently fails when running `yarn test:acceptance`

File: test/acceptance/nightwatch.conf.js [hunk 5]
```diff
@@ -1,18 +1,24 @@
 require('dotenv').config()
 const seleniumServer = require('selenium-server')
 const chromeDriver = require('chromedriver')
+const { readdirSync } = require('fs')
+
+const pageObjects =
```
Comment (feedmypixel) [inline test/acceptance/nightwatch.conf.js]: sorry my bad after checking it out feels like these are `pageObjectPaths`?

File: test/acceptance/nightwatch.conf.js [hunk 6]
```diff
@@ -1,18 +1,24 @@
...
+  readdirSync('./features')
+    .filter((featuresFolderName) => { return featuresFolderName !== 'step-definitions' })
```
Comment (feedmypixel) [inline test/acceptance/nightwatch.conf.js]: formatting is a little funky. Would suggest:
```
const pageObjectPaths = readdirSync('test/acceptance/features')
  .filter((featuresFolderName) => {
    return featuresFolderName !== 'step-definitions'
  })
  .map((appName) => {
    return `test/acceptance/features/${appName}/page-objects`
  })
```
or

```
const pageObjectPaths = readdirSync('test/acceptance/features')
  .filter((featuresFolderName) => featuresFolderName !== 'step-definitions')
  .map((appName) => `test/acceptance/features/${appName}/page-objects`)
```

File: test/acceptance/nightwatch.conf.js (hunk 1 above)
Comment (tyom) [reply test/acceptance/nightwatch.conf.js]: Could just do `const pageObjects = glob.sync('test/acceptance/features/**/page-objects')` with `glob`.

**Pull request merged**